packages/xo-web/src/common/xo/index.js

```diff
@@ -821,14 +821,43 @@ export const setRemoteSyslogHost = (host, syslogDestination) =>
 export const setRemoteSyslogHosts = (hosts, syslogDestination) =>
   Promise.all(map(hosts, host => setRemoteSyslogHost(host, syslogDestination)))
 
-export const restartHost = (host, force = false, suspendResidentVms = false) =>
+export const restartHost = (host, force = false, suspendResidentVms = false, forceSmartReboot = false) =>
   confirm({
     title: _('restartHostModalTitle'),
     body: _('restartHostModalMessage'),
   }).then(
     () =>
-      _call('host.restart', { id: resolveId(host), force, suspendResidentVms })
+      _call('host.restart', { id: resolveId(host), force, suspendResidentVms, forceSmartReboot })
         .catch(async error => {
+          if (
+            suspendResidentVms &&
+            !forceSmartReboot &&
+            operationBlocked.is(error, {
+              code: 'suspend',
+            })
+          ) {
+            await confirm({
+              body: (
+                <p>
+                  <Icon icon='alarm' /> {_('forceSmartRebootHost', { nVms: error.data.objectId.length })}
+                </p>
+              ),
+              title: _('restartHostModalTitle'),
+            })
+            forceSmartReboot = true
+            return _call('host.restart', {
+              id: resolveId(host),
+              force,
+              suspendResidentVms,
+              forceSmartReboot,
+            })
+          }
+          throw error
+        })
+        .catch(async error => {
+          if (error === undefined) {
+            return
+          }
```

Comment:
I don't think it makes much difference with the current implementation but technically, `undefined` should be thrown and not returned, so that it's properly interpreted as "the action was canceled" instead of "the action was run and returned `undefined`".